c10/util/MPMCQueue.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * A simple multi-producer, multi-consumer queue.
+ *
+ * This is a wrapper around std::deque that provides
+ * readIfNotEmpty and writeIfNotFull using std mutexes.
```

Comment:
nit: I guess the point is for them to be thread safe. Otherwise we wouldn't need to have a custom class for this and an API designed to be easily thread-safe?
I would mention that here as a requirement in case someone changes it later. In particular, it is very important to not provide access to the queue_ directly!